app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java

```diff
@@ -308,6 +314,33 @@ public boolean onContextItemSelected(MenuItem item) {
             case R.id.move_to_top_item:
                 DBWriter.moveQueueItemToTop(getActivity(), selectedItem.getId(), true);
                 return true;
+            case R.id.mark_read_item:
+                DBWriter.markItemRead(getActivity(), selectedItem, true, false);
+                selectedItem.setRead(true);
+                if(GpodnetPreferences.loggedIn()) {
```

Comment:
That you're doing the Gpodder work here made me realize a couple of things:

1. We should probably have some central location that is handling the Gpodder stuff so that we don't forget about it
2. We're probably not handling it everywhere we should. Like when marking all items as read or swiping to mark something as read.